guide/src/async_python_from_rust.md

```diff
@@ -0,0 +1,186 @@
+# Calling async Python functions in Rust code
+
+In the previous chapter, we learnt how to call Python functions in Rust code. Now, we will look at two ways to call Python async functions from Rust code.
+
+## Async Functions
+
+Before we start, let us take a look at the following snippet.
+
+```python
+async def fx():
+    return "hello, world"
+```
+
+Above, we have defined an async function. How we send it to Rust code will make a difference.
+
+In python, even an async function is treated like a regular function until called.
+e.g.
+
+```python
+print(type fx) # this will output the type function
+print(type fx()) # this will output the type of a coroutine
+```
+
+The native object `PyAny` can be used to call the Python functions in Rust code.
+
+Now, let us take a look at how will we share the Python function in our rust code.
```

Comment:
Can we add some more detail here about what we will do, so that the reader can feel a sense of direction?